arbnode/inbox_reader.go

```diff
@@ -137,6 +142,15 @@ func (ir *InboxReader) run(ctx context.Context) error {
 	newHeaders, unsubscribe := ir.l1Reader.Subscribe(false)
 	defer unsubscribe()
 	blocksToFetch := uint64(100)
+	seenBatchCount := uint64(0)
+	seenBatchCountStored := uint64(math.MaxUint64)
+	storeSeenBatchCount := func() {
+		if seenBatchCountStored != seenBatchCount {
+			atomic.StoreUint64(&ir.lastSeenBatchCount, seenBatchCount)
+			seenBatchCountStored = seenBatchCount
```

Comment:
We can just use `ir.lastSeenBatchCount` here instead of `seenBatchCountStored`. We also don't need an atomic load because we're the exclusive writer.

Reply:
atomic is used because there is no syn between reader thread and writer thread. In relevant systems - uint64 access is actually atomic even with normal loads/stores - but I think it's still worthwhile, if only to emphasize that.